src/java.base/share/classes/jdk/internal/classfile/impl/ClassReaderImpl.java

```diff
@@ -242,12 +270,20 @@ public double readDouble(int p) {
 
     @Override
     public byte[] readBytes(int p, int len) {
-        return Arrays.copyOfRange(buffer, p, p + len);
+        try {
+            return Arrays.copyOfRange(buffer, p, p + len);
+        } catch (IndexOutOfBoundsException e) {
+            throw outOfBoundsError(e);
+        }
     }
 
     @Override
     public void copyBytesTo(BufWriter buf, int p, int len) {
-        buf.writeBytes(buffer, p, len);
+        try {
+            buf.writeBytes(buffer, p, len);
```

Comment:
`java.lang.classfile.BufWriter` doesn't specify any `@throws` for its `writeXXX` methods. Should it be specified (of course in a separate PR)?

Reply:
That is a good point, thanks!

Reply:
I've created [JDK-8322847](https://bugs.openjdk.org/browse/JDK-8322847)